src/Bicep.Core/Syntax/ImportSpecification.cs

```diff
@@ -46,35 +48,57 @@ private ImportSpecification(string name, string version, bool isValid, TextSpan
-                    var (name, version, isValid) = Parse(value);
-                    var span = isValid ? new TextSpan(stringSyntax.Span.Position + 1, name.Length) : stringSyntax.Span;
-
-                    return new ImportSpecification(name, version, isValid, span);
-
-                case SkippedTriviaSyntax trivia:
-                    return new ImportSpecification(trivia.TriviaName, trivia.TriviaName, false, trivia.Span);
+                StringSyntax stringSyntax when stringSyntax.TryGetLiteralValue() is { } value
+                    => CreateFromStringSyntax(stringSyntax, value),
+                SkippedTriviaSyntax trivia
+                    => new ImportSpecification(
+                        trivia.TriviaName,
+                        trivia.TriviaName,
+                        trivia.TriviaName,
+                        false,
+                        trivia.Span),
+                _
+                    => new ImportSpecification(
+                        LanguageConstants.ErrorName,
+                        LanguageConstants.ErrorName,
+                        LanguageConstants.ErrorName,
+                        false,
+                        specificationSyntax.Span)
+            };
+        }
 
-                default:
-                    return new ImportSpecification(LanguageConstants.ErrorName, LanguageConstants.ErrorName, false, specificationSyntax.Span);
+        public SyntaxBase ToPath()
+        {
+            if (!this.IsValid)
+            {
+                return new SkippedTriviaSyntax(this.Span, Enumerable.Empty<SyntaxBase>());
             }
+            return SyntaxFactory.CreateStringLiteral(this.UnexpandedPath);
         }
 
-        private static (string Name, string Version, bool IsValid) Parse(string value)
+        private static ImportSpecification CreateFromStringSyntax(StringSyntax stringSyntax, string value)
         {
-            var match = SpecificationPattern.Match(value);
-
+            var match = BareSpecification.Match(value);
             if (!match.Success)
             {
-                return (LanguageConstants.ErrorName, LanguageConstants.ErrorName, false);
+                return new(
+                 LanguageConstants.ErrorName,
+                 LanguageConstants.ErrorName,
+                 LanguageConstants.ErrorName,
+                 false,
+                 stringSyntax.Span);
             }
 
             var name = match.Groups["name"].Value;
             var version = match.Groups["version"].Value;
+            var parts = value.Split('@');
+            var unexpandedRepositoryAddress = parts[0];
 
-            return (name, version, true);
+            var span = new TextSpan(stringSyntax.Span.Position + 1, unexpandedRepositoryAddress.Length);
+            var unexpandedArtifactAddress = value.Replace('@', ':'); // we use the same format as module references so we can reuse module dispatcher logic.
```

Comment:
This is an interesting finding. I'm wondering if it's worth having a quick discussion on the pros/cons of having a syntax with `@` to represent the version (clearer), vs having a syntax that fits with the existing module path resolution logic (familiar to existing users)

Reply:
I agree, it's a good point but I worry it's not in scope of this PR. Would you like me to do a write up and bring this up in the bicep discussions forum or do you prefer to do it given it's an insight you have identified?

Reply:
> I agree, it's a good point but I worry it's not in scope of this PR.

Yeah I think it's fine to create a task to track.